File: app/src/main/java/fr/corenting/convertisseureurofranc/convert/USA.java
package fr.corenting.convertisseureurofranc.convert;

import android.content.Context;

import fr.corenting.convertisseureurofranc.R;

public class USA extends ConvertAbstract {

    public USA(Context context) {
        this.context = context;
        loadValuesFromCSV(R.raw.us_values);
    }

    public double convertFunction(int yearOfOrigin, int yearOfResult, float amount) {
        if (yearOfOrigin == yearOfResult) return amount;
        double multiplier = values.get(yearOfResult) / values.get(yearOfOrigin);
        return amount * multiplier;
    }

    public String getCurrencyFromYear(int year) {
        return context.getString(R.string.dollars);
    }
}

File: app/src/main/java/fr/corenting/convertisseureurofranc/convert/USA.kt
package fr.corenting.convertisseureurofranc.convert

import android.content.Context

import fr.corenting.convertisseureurofranc.R

class USA(context: Context) : ConvertAbstract() {

    init {
        this.setContext(context)
        loadValuesFromCSV(R.raw.us_values)
    }

    override fun convertFunction(yearOfOrigin: Int, yearOfResult: Int, amount: Float): Double {
        if (yearOfOrigin == yearOfResult) return amount.toDouble()
        val multiplier = getValues().get(yearOfResult) / getValues().get(yearOfOrigin)
        return amount * multiplier
    }

    override fun getCurrencyFromYear(year: Int): String {
        return getContext().getString(R.string.dollars)
    }
}
